WebConfTech/website-2019
# Video and photos

## What does this PR do?
Modifies the main page to show the webconf summary video and add the photos section. 

### Notes:
- Upgraded dependencies minor versions (including gatsby) to be able to use some features and to fix the security warning issue on the repo.
- Implemented a Gatsby plugin to bring photos from the facebook album where we published our selection. Let me know if we want to use a different source for the photos.
- Hardcoded some resolution values on the code since we need pixels numbers and the constants use `rem` let me know if I should change it or it's okay as it is.
- Had to implement two separate photo picker components depending on the browser resolution since their structure and behaviour are pretty different.
- Did my best to add some lazy loading/preloading to avoid overfetching but still trying to preserve a good UX.
- Just noticed a weird "jump" when switching between pages on the photo's picker for desktop, not sure why it's happening, need to investigate.
- Something to take into account: browsers have some weird policies on autoplaying videos nowaday, that may affect the home's fullscreen video. I'm not sure if we can ask the user for autoplay permission or adding some way for the user to start the video themselves. We may have to research that.

## How should it be tested manually?
Go to the deploy and check the home and photos section's design against the designs at https://xd.adobe.com/spec/bddd016f-156f-466b-5d06-8c8128fc6216-cbec/.
Also make sure to check that the behaviour of the photo picker and it's lazy loading/prefetching works as expected.
